### PR TITLE
Fix proposal details for small devices

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Optimise nns proposals for middle-size android devices (eg.Galaxy S21).

# Changes

- apply sns proposer UI for nns proposer
- fix broken ellipsis in proposal card

# Tests

| Before | After |
|--------|--------|
| Screenshot 2023-11-22 at 10 58 12 | Screenshot 2023-11-22 at 10 59 29 |
| image | image | 

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: frontend/src/lib/components/ui/Hash.svelte [hunk 1]
```diff
@@ -10,14 +11,24 @@
...
   export let className: string | undefined = undefined;
   export let splitLength: number | undefined = undefined;
+  export let tooltipTop: boolean | undefined = undefined;
+
+  const dispatcher = createEventDispatcher();
 
   let shortenText: string;
   $: shortenText = shortenWithMiddleEllipsis(text, splitLength);
+
+  $: console.log("text", text, shortenText);
```
Comment (peterpeterparker) [inline frontend/src/lib/components/ui/Hash.svelte:21]: to be removed

File: frontend/src/lib/components/ui/Hash.svelte [hunk 2]
```diff
@@ -10,14 +11,24 @@
...
 
 <span data-tid="hash-component">
-  <Tooltip {id} {text}>
-    <svelte:element this={tagName} data-tid={testId} class={className}>
+  <Tooltip top={tooltipTop} {id} {text}>
+    <svelte:element
+      this={tagName}
+      data-tid={testId}
+      class={className}
+      on:click|stopPropagation={() => dispatcher("nnsTextClick")}
```
Comment (peterpeterparker) [inline frontend/src/lib/components/ui/Hash.svelte:30]: Should we rename the event to `nnsHash` ?

File: frontend/src/lib/components/proposal-detail/ProposalSystemInfoProposerEntry.svelte [hunk 3]
```diff
@@ -2,15 +2,21 @@
...
+  import Hash from "$lib/components/ui/Hash.svelte";
+  import { Html, KeyValuePairInfo } from "@dfinity/gix-components";
 
   export let proposer: NeuronId | undefined;
 
   let modalOpen = false;
-
   // TODO: For simplicity reason, currently we do not display the proposer details if not signed-in. We might want to improve the display of these information in the future.
+
+  const click = () => {
```
Comment (peterpeterparker) [inline frontend/src/lib/components/proposal-detail/ProposalSystemInfoProposerEntry.svelte:15]: maybe rename to "showVotingHistoryModal" or something like to makes it more explicit?

File: frontend/src/lib/components/ui/Hash.svelte (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/ui/Hash.svelte]: Thanks!

File: frontend/src/lib/components/proposal-detail/ProposalSystemInfoProposerEntry.svelte (hunk 3 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/proposal-detail/ProposalSystemInfoProposerEntry.svelte]: Good point, renamed.

File: frontend/src/lib/components/ui/Hash.svelte (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/ui/Hash.svelte]: Done 👍

**Pull request merged**